### PR TITLE
Basic error handling with a pretty snackbar

### LINKED ISSUE #205: [Frontend] - Sign up the same email a second time should display an error to the user
Backend already handles this scenario:

Currently the page does not do anything when getting 400 response as above but we ideally should display the user an error saying that his account was already created

Acceptance Criteria:
When the user is trying to sign up with the same email, the page should display en error message.

### PR DESCRIPTION
**Pull request by ds161833**
This PR displays error message on API calls if returned 4xx or 5xx
image
image

https://user-images.githubusercontent.com/22376976/105072075-69475500-5a53-11eb-98e1-5f59a5b6a1d0.mp4

https://user-images.githubusercontent.com/22376976/105072078-6a788200-5a53-11eb-8d3a-557b3176f362.mp4

### CODE REVIEW

**Review by Vatika1 — APPROVED** [review]
looks pretty :P

File: Client/src/Components/SnackbarUtils.tsx [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+  warning(msg: string) {
+    this.toast(msg, 'warning')
+  },
+  info(msg: string) {
+    this.toast(msg, 'info')
+  },
+  error(msg: string) {
+    this.toast(msg, 'error')
+  },
+  toast(msg: string, variant: VariantType = 'default') {
```
Comment (GD-Dheer) [inline Client/src/Components/SnackbarUtils.tsx:36]: Generic snack messages A+

File: Client/src/Remote/RemoteHelper.tsx [hunk 2]
```diff
@@ -31,11 +31,25 @@ const fetchRemote = async (url: string, method: string, data: any = {}): Promise
   setData(request, data)
 
   try {
-    return fetch(url, request)
+    const response = await fetch(url, request)
+
+    if (response.status.toString().charAt(0) == '5' && !response.body) {
```
Comment (GD-Dheer) [inline Client/src/Remote/RemoteHelper.tsx]: The returned response.body should be filled on error 500 with an error message

File: Client/src/Remote/RemoteHelper.tsx [hunk 3]
```diff
@@ -14,15 +16,13 @@ const requestBase: RequestInit = {
...
 
 export const get = async (route: string): Promise<any> => {
   const url = route
 
-  const response = await fetchRemote(url, 'GET')
-  return response.json()
+  return fetchRemote(url, 'GET')
 }
 
 const fetchRemote = async (url: string, method: string, data: any = {}): Promise<Response> => {
```
Comment (GD-Dheer) [inline Client/src/Remote/RemoteHelper.tsx:28]: Is it better to make the 3rd param here optional with (?) ?

File: Client/src/Remote/RemoteHelper.tsx [hunk 4]
```diff
@@ -14,15 +16,13 @@ const requestBase: RequestInit = {
...
-  return response.json()
+  return fetchRemote(url, 'POST', data)
 }
 
 export const get = async (route: string): Promise<any> => {
   const url = route
 
-  const response = await fetchRemote(url, 'GET')
-  return response.json()
+  return fetchRemote(url, 'GET')
```
Comment (GD-Dheer) [inline Client/src/Remote/RemoteHelper.tsx:25]: Overloading A+

File: Client/src/Remote/RemoteHelper.tsx [hunk 5]
```diff
@@ -31,11 +31,25 @@ const fetchRemote = async (url: string, method: string, data: any = {}): Promise
...
+      SnackbarUtils.error('Server Unavailable')
+      Promise.resolve(null)
+    }
+
+    const message = await response.json()
+    if (response.status.toString().charAt(0) == '2') {
+      return message
+    }
+
+    if (response.status.toString().charAt(0) == '4') {
```
Comment (GD-Dheer) [inline Client/src/Remote/RemoteHelper.tsx:47]: I like the way you mapped the error status to the snack error type here

File: Client/src/App.tsx [hunk 6]
```diff
@@ -2,6 +2,8 @@ import React, { Dispatch, SetStateAction, useState } from 'react'
 
 import { IUserAccountModel } from './Remote/Models/IUserAccountModel'
 import NavigationMenu from './Components/NavigationMenu'
+import { SnackbarProvider } from 'notistack'
```
Comment (GD-Dheer) [inline Client/src/App.tsx:5]: We'll have to remember that we need to write a little something on why we are using 3rd party libraries. This one has sweet snack bars for us

**Review by GD-Dheer — APPROVED** [review]
Just need to double check the status 500 code. Other than that its a nice generic fix

File: Client/src/App.tsx (hunk 6 above)
Comment (ds161833) [reply Client/src/App.tsx:5]: added to the todo list for release 2

**Review by ds161833 — COMMENTED** [review]
> Just need to double check the status 500 code. Other than that its a nice generic fix

After a live discussion we decided to display a generic 'Server Unavailable' error on any 500 error

**Pull request merged**